wallet/src/de/schildbach/wallet/ui/invite/InviteHandler.kt

```diff
@@ -62,24 +60,28 @@ class InviteHandler(val activity: FragmentActivity, private val analytics: Analy
             val mainTask = getMainTask(activity)
             if (walletApplication.wallet != null) {
                 // if wallet exists, go to the Home Screen
-                mainTask.startActivity(activity.applicationContext, MainActivity.createIntent(activity), null)
+                val intent = MainActivity.createIntent(activity)
+                mainTask?.startActivity(activity.applicationContext, intent, null)
+                    ?: activity.startActivity(intent)
             } else {
-                mainTask.startActivity(activity.applicationContext, OnboardingActivity.createIntent(activity), null)
+                val intent = OnboardingActivity.createIntent(activity)
+                mainTask?.startActivity(activity.applicationContext, intent, null)
+                    ?: activity.startActivity(intent)
             }
             activity.finish()
         }
 
         private fun handleMoveToFront(activity: FragmentActivity) {
             activity.setResult(Activity.RESULT_CANCELED)
             val mainTask = getMainTask(activity)
-            mainTask.moveToFront()
+            mainTask?.moveToFront()
             activity.finish()
         }
     }
 
-    fun handle(inviteResource: Resource<InvitationLinkData>, silentMode: Boolean = false) {
+    fun handle(inviteResource: Resource<InvitationLinkData>, silentMode: Boolean = false, successFunction: ((InvitationLinkData) -> Unit)? = null) {
```

Comment:
This is why I'm not a fan of `Resource<>` structure. This method would be much simpler as a `suspend fun handle(): InvitationLinkData?`

Reply:
`handle` shows a loading dialog and cancels it based on the Resource (loading, success, error, etc), which is why I left it this way.  how else could we pass this extra information or handle it?